### PR TITLE
mathgl: init at 8.0.1

### PR DESCRIPTION
**Pull request by GabrielDougherty**
###### Description of changes

Adds a package for MathGL, which is a library for scientific data visualization. Homepage is here: https://mathgl.sourceforge.net/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mathgl/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  version = "8.0.1";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/mathgl/mathgl-${version}.tar.gz";
+    sha256 = "sha256-yoS/lIDDntMRLpIMFs49jyiYaY9iiW86V3FBKGIqVao=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    zlib
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mathgl/default.nix:20]: Libraries go to buildInputs

File: pkgs/development/libraries/mathgl/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    libpng
+    libGL
+  ];
+
+  meta = with lib; {
+    description = "A library for scientific data visualization";
+    homepage = "https://https://mathgl.sourceforge.net/";
+    license = with licenses; [ gpl3 lgpl3 ];
+    platforms = platforms.linux ++ platforms.windows;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mathgl/default.nix:29]: Did you try building it for windows?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/mathgl/default.nix (hunk 2 above)
Comment (GabrielDougherty) [reply pkgs/development/libraries/mathgl/default.nix:29]: No, but the underlying project mathgl supports Windows.

File: pkgs/development/libraries/mathgl/default.nix (hunk 1 above)
Comment (GabrielDougherty) [reply pkgs/development/libraries/mathgl/default.nix:20]: Done

File: pkgs/development/libraries/mathgl/default.nix (hunk 2 above)
Comment (GabrielDougherty) [reply pkgs/development/libraries/mathgl/default.nix:29]: Would you rather this say `platforms = platforms.linux;`? I am not aware of best practices here.

Comment (GabrielDougherty) [reply pkgs/development/libraries/mathgl/default.nix]: Updated to leave out Windows.

File: pkgs/development/libraries/mathgl/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "mathgl";
+  version = "8.0.1";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/mathgl/mathgl-${version}.tar.gz";
+    sha256 = "sha256-yoS/lIDDntMRLpIMFs49jyiYaY9iiW86V3FBKGIqVao=";
+  };
+
+  buildInputs = [
+    cmake
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mathgl/default.nix:19]:
```suggestion
  nativeBuildInputs = [
    cmake
  ];
    
  buildInputs = [
```

**Pull request merged**